src/drivers/fluid_haiku.cpp

```diff
@@ -0,0 +1,137 @@
+/* Haiku MediaKit Driver for FluidSynth
+ *
+ * Copyright (C) 2019  Gerasim Troeglazov
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * as published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA
+ */
+
+
+#include <MediaKit.h>
+#include <SupportKit.h>
+
+extern "C" {
+#include "fluid_synth.h"
+#include "fluid_adriver.h"
+#include "fluid_settings.h"
+}
+
+typedef struct
+{
+    fluid_audio_driver_t driver;
+
+    fluid_synth_t *synth;
+    fluid_audio_callback_t write_ptr;
+
+    int frame_size;
+	
+	BSoundPlayer *m_player;
+
+} fluid_haiku_audio_driver_t;
+
+
+static void playerProc(void *cookie, void *buffer, size_t len, const media_raw_audio_format &format)
+{
+    fluid_haiku_audio_driver_t *dev = (fluid_haiku_audio_driver_t *)cookie;
+    len /= dev->frame_size;
+    dev->write_ptr(dev->synth, len, buffer, 0, 2, buffer, 1, 2);
+}
+
+void fluid_haiku_audio_driver_settings(fluid_settings_t *settings)
+{
+    fluid_settings_register_str(settings, "audio.haiku.device", "default", 0);
+    fluid_settings_add_option(settings, "audio.haiku.device", "default");
+}
+
+fluid_audio_driver_t *
+new_fluid_haiku_audio_driver(fluid_settings_t *settings, fluid_synth_t *synth)
+{
+    fluid_haiku_audio_driver_t *dev = NULL;
+    fluid_audio_callback_t write_ptr;
+    
+    double sample_rate;
+    int period_size;
+    int periods;
+    int sample_size;
+    
+    char *device;
+    const char *dev_name;
```

Comment:
Looks like these two variables were intended to be used with `audio.haiku.device`, but they are unused. If so, pls. remove them, and then also remove the `audio.haiku.device` setting.

Reply:
Seems like some unimplemented stuff yeah.